plasmapy/formulary/paschen.py

```diff
@@ -0,0 +1,279 @@
+            "Al": 0.053,
+            "Cu": 0.02,
+            "Ni": 0.031,
+            "Pt": 0.023,
+            "C": None,
+            "W": 0.045,
+            "Fe": 0.022,
+        },
+        "Ar": {
+            "Al": 0.12,
+            "Cu": 0.058,
+            "Ni": 0.058,
+            "Pt": 0.058,
+            "C": None,
+            "W": None,
+            "Fe": 0.058,
+        },
+        "Kr": {
+            "Al": None,
+            "Cu": None,
+            "Ni": None,
+            "Pt": None,
+            "C": None,
+            "W": None,
+            "Fe": None,
+        },
+        "Xe": {
+            "Al": None,
+            "Cu": None,
+            "Ni": None,
+            "Pt": None,
+            "C": None,
+            "W": None,
+            "Fe": None,
+        },
+    }
+
+    # Check if the asked gas and electrode material is supported
+    resg = gas in gases
+    rese = electrode in materials
+
+    # If the gas is supported get the constants A and B
+    if resg:
+        A = paschen_constants[gas][0]
+        B = paschen_constants[gas][1]
+
+        #   Get the townsend_gamma coefficient for the the gas/electrode combination
+        if rese:
+            gam = townsend_gamma[gas]
+            gn = gam[electrode]
+            #       Test if townsend_gamma exists for the demanded gas/electrode configuration
+            #       If not a default townsend_gamma value of 0.01 is taken
+
+            if gn is None:
+                gn = 0.01
+        else:
+
+            #       If the electrode material is not supportes set townsend_gamma to default = 0.01
+            gn = 0.01
+
+        #   Create output dir {const}
+        const = {"A": A, "B": B, "gam": gn}
+        return const
+
+    # If gas is not supported set const=None
+    else:
+        const = None
+        return const
+
+
+def breakdown_voltage(distance, pressure, A, B, gam):
+    r"""
+    Calculate the breakdown voltage V according to the Paschen law
+
+                         𝑉=𝐵𝑝𝑑/𝑙𝑛(𝐴𝑝𝑑/𝑙𝑛(1+1/𝛾))
+
+
+    Parameters
+    ----------
+
+    distance:  floating
+    electrode distance
+
+    pressure:  list
+    Gas pressure
+
+    A:         floating
+    Paschen constant A
+
+    B:         floating
+    Paschen constant B
+
+    gam:       floating
+    Second Townsend coefficient
+
+    The parameters A,B and gam can be obtain for some typical gases and electrode materials
+    from the function get_paschen_constants, other values can be also be introduced
+
+
+    Return
+    ------
+    [pd,breakdown_voltage] :  list
+
+    [p*d value, Paschen breakdown voltage]
+
+
+    Examples
+    --------
+
+    >>> breakdown_voltage(0.1,[100,200],11,135,0.058)
+    [(10.0, 371.44065672260245), (20.0, 623.8960061470598)]
+
+    """
+
+    #  Calculate breakdown voltage according to the Paschen law
+    x = list(distance * p for p in pressure)
+    g = np.log(1 + (1 / gam))
+    vb = list((pd, B * pd / (np.log((A * pd) / g))) for pd in x)
+    return vb
```

Comment:
It might be easier and faster to require a numpy array rather than the list, and return a 2-tuple of numpy arrays rather than a list of 2-tuples.